net.codeindustry.MasterPDFEditor.yaml

```diff
@@ -9,6 +9,7 @@ finish-args:
   - --share=network
   - --socket=x11
   - --socket=wayland
+  - --env=TMPDIR=/var/tmp
```

Comment:
I'm happy that this works, but it's unfortunate that it's not a directory private to the app.
Can you try to launch the app with `TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"`, e.g. through the desktop file:  `Exec=env TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID" masterpdfeditor5 %f` ?

Reply:
I guess the "Exec" must include flatpak.

I ended up with this but it doesn't run:

```
$ cat net.codeindustry.MasterPDFEditor.desktop 
[Desktop Entry]
Name=Master PDF Editor 5
Comment=Edit PDF files
Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command="env TMPDIR='$XDG_RUNTIME_DIR/app/$FLATPAK_ID' masterpdfeditor5" --file-forwarding net.codeindustry.MasterPDFEditor @@ %f @@
Terminal=false
Icon=net.codeindustry.MasterPDFEditor
Type=Application
Categories=Office;Graphics;
MimeType=application/pdf;application/x-bzpdf;application/x-gzpdf;
X-Flatpak=net.codeindustry.MasterPDFEditor
```

Reply:
You can't use `--env=...` with environment variables, you'd have to instead make a small wrapper script around the main binary. (There are quite a few other Flatpaks that do this, an example of a wrapper script is [here](https://github.com/flathub/im.riot.Riot/blob/master/im.riot.Riot.yaml#L110-L114)).

Reply:
I have modified the manifest as follows but that still doesn't take care of the bug:
```
app-id: net.codeindustry.MasterPDFEditor
runtime: org.kde.Platform
runtime-version: '5.15'
sdk: org.kde.Sdk
command: masterpdfeditor5wrapper
finish-args:
  - --device=dri
  - --share=ipc
  - --share=network
  - --socket=x11
  - --socket=wayland
modules:
  - name: sane-backends
    sources:
      - type: archive
        url: https://gitlab.com/sane-project/backends/uploads/9e718daff347826f4cfe21126c8d5091/sane-backends-1.0.28.tar.gz
        sha256: 31260f3f72d82ac1661c62c5a4468410b89fb2b4a811dabbfcc0350c1346de03
    post-install:
      - echo 127.0.0.1 > /app/etc/sane.d/net.conf
      - echo net > /app/etc/sane.d/dll.conf

  - name: masterpdfeditor5
    buildsystem: simple
    build-commands:
      - mkdir -p /app/bin
      - ln -s /app/{extra,bin}/masterpdfeditor5
      - install -Dm 755 apply_extra.sh /app/bin/apply_extra
      - install -Dm 644 net.codeindustry.MasterPDFEditor.metainfo.xml -t /app/share/metainfo
      - install -Dm 644 net.codeindustry.MasterPDFEditor.desktop -t /app/share/applications
      - install -Dm 644 net.codeindustry.MasterPDFEditor.png -t /app/share/icons/hicolor/128x128/apps
      - install masterpdfeditor5wrapper.sh /app/bin/masterpdfeditor5wrapper
    sources:
      - type: extra-data
        url: https://code-industry.net/public/master-pdf-editor-5.8.06-qt5.x86_64.tar.gz
        filename: master-pdf-editor.tar.gz
        sha256: 7fd4d03e97d1e04499846ebc516b967ec5777bbdf6e794deee196376a3849b17
        size: 15975595
        x-checker-data:
          type: html
          url: https://code-industry.net/free-pdf-editor/
          version-pattern: master-pdf-editor-([0-9.]+)-qt5\.x86_64\.tar\.gz
          url-template: https://code-industry.net/public/master-pdf-editor-$version-qt5.x86_64.tar.gz
      - type: script
        dest-filename: apply_extra.sh
        commands:
          - tar --strip-components=1 -xf master-pdf-editor.tar.gz
          - rm master-pdf-editor.tar.gz
      - type: file
        path: net.codeindustry.MasterPDFEditor.metainfo.xml
      - type: file
        path: net.codeindustry.MasterPDFEditor.desktop
      - type: file
        path: net.codeindustry.MasterPDFEditor.png
      - type: script
        dest-filename: masterpdfeditor5wrapper.sh
        commands:
          - export TMPDIR="$XDG_RUNTIME_DIR/app/${FLATPAK_ID:-net.codeindustry.MasterPDFEditor}"
          - /app/bin/masterpdfeditor5 "$@"
 ```

Reply:
I did it! I'm so happy :) Gonna resubmit